src/layer.js

```diff
@@ -145,50 +145,63 @@ export default class Layer {
    * @return {Promise} [will resolve once image src has been retrieved and dimensions calculated]
    */
   calculateSize() {
-    // Create image element
-    this.imageEl = new Image();
-    this.imageEl.crossOrigin = 'Anonymous';
-    this.imageEl.src = this.data.image;
+    var _this = this;
+    var img = new Image();
+    img.crossOrigin = 'Anonymous';
+    return (new Promise((resolve, reject) => {
+      img.onerror = (err) => reject(err);
```

Comment:
remove the parenthesis off this argument as it will fail eslint tests, if 1 argument is passed you use
`img.onerror = err => reject(err);`